src/interfaces/iPeer.ts

```diff
@@ -0,0 +1,39 @@
+import { Message } from "../models/message";
+
+/**
+ * Peer interface.
+ */
+export interface IPeer {
+    /**
+     * Register custom receiver handlers.
+     *
+     * @param messageType - The messageType that the receiver handles
+     * @param implementation - The implementation of the receiver handler
+     *                         The implementation should be a void that receives the following arguments:
+     *                              message - The message
+     *                              sender - The IP of the message sender
+     */
+    registerReceiveHandlerImpl(messageType: string, implementation: (message: Message, sender: string) => void): void;
```

Comment:
If I'm reading this correct this method implements its own implement parameter. Or why is the method itself called Impl?